src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -196,8 +197,11 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
 
       // Check if any place territories exist
       final Map<Territory, ProPurchaseTerritory> purchaseTerritories = purchaseUtils.findPurchaseTerritories(player);
-      if (purchaseTerritories.isEmpty()) {
-        ProLogger.info("No possible place territories owned so exiting purchase logic");
+      final List<Territory> possibleFactoryTerritories =
```

Comment:
I notice now the "NoInfra" has a "invert" at the end.

@ron-murhammer , please make sure you intended to call NoInfra rather than the hasinfra and fix as necessary.